File: Actly.Client/src/app/services/participation.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { ParticipationDto } from '../models/participation';
import { Observable, of } from 'rxjs';
import { map, catchError } from 'rxjs/operators';

@Injectable({
  providedIn: 'root'
})
export class ParticipationService {
  private apiUrl = 'http://localhost:5126/api/Participation';

  constructor(private http: HttpClient) {}

  checkParticipation(userId: number, eventId: number): Observable<boolean> {
    return this.http.get<any>(`${this.apiUrl}/user/${userId}/event/${eventId}`).pipe(
      map(response => !!response),
      catchError(() => of(false)) 
    );
  }

  joinEvent(participation: ParticipationDto): Observable<any> {
    return this.http.post(this.apiUrl, participation);
  }

  leaveEvent(userId: number, eventId: number): Observable<any> {
    return this.http.delete(`${this.apiUrl}/${userId}/${eventId}`);
  }
}

File: Actly.API/Models/Participation.cs
using System.ComponentModel.DataAnnotations;

namespace Actly.API.Models
{
    public class Participation
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        public required User? User { get; set; }

        [Required]
        public int EventId { get; set; }

        public required Event? Event { get; set; }
        public bool Attended { get; set; }

        [Required]
        public DateTime JoinedAt { get; set; } = DateTime.UtcNow;
    }
}